test/locks.js

```diff
@@ -39,231 +47,219 @@ describe('Locks', function() {
+    assert(apos.modules['apostrophe-locks-1']);
+    assert(apos.modules['apostrophe-locks-2']);
+    assert(apos.modules['apostrophe-locks-3']);
   });
 
-  it('should allow a single lock without contention uneventfully', function(done) {
-    let locks = apos.modules['apostrophe-locks'];
-    return async.series([ lock, unlock ], function(err) {
-      assert(!err);
-      done();
-    });
-    function lock(callback) {
-      return locks.lock('test', callback);
-    }
-    function unlock(callback) {
-      return locks.unlock('test', callback);
+  it('cleanup', async function() {
+    try {
+      const response = await apos.locks.db.remove({});
+
+      assert(response.result.ok === 1);
+    } catch (e) {
+      assert(false);
     }
   });
 
-  it('should allow two differently-named locks uneventfully', function(done) {
-    let locks = apos.modules['apostrophe-locks'];
-    return async.series([ lock1, lock2, unlock1, unlock2 ], function(err) {
-      assert(!err);
-      done();
-    });
-    function lock1(callback) {
-      return locks.lock('test1', callback);
-    }
-    function unlock1(callback) {
-      return locks.unlock('test1', callback);
-    }
-    function lock2(callback) {
-      return locks.lock('test2', callback);
-    }
-    function unlock2(callback) {
-      return locks.unlock('test2', callback);
+  it('should allow a single lock without contention uneventfully', async function() {
+    const locks = apos.modules['apostrophe-locks'];
+
+    try {
+      await locks.lock('test');
+
+      await locks.unlock('test');
+    } catch (e) {
+      assert(false);
     }
+
+    // function lock(callback) {
+    //   return locks.lock('test', callback);
+    // }
+    // function unlock(callback) {
+    //   return locks.unlock('test', callback);
+    // }
   });
 
-  it('should flunk a second lock by the same module', function(done) {
-    let locks = apos.modules['apostrophe-locks'];
-    return async.series([ lock, lockAgain, unlock, unlockAgain ], function(err) {
-      assert(!err);
-      done();
-    });
-    function lock(callback) {
-      return locks.lock('test', callback);
-    }
-    function lockAgain(callback) {
-      return locks.lock('test', function(err) {
-        // SHOULD fail
-        assert(err);
-        return callback(null);
-      });
+  it('should allow two differently-named locks uneventfully', async function() {
+    const locks = apos.modules['apostrophe-locks'];
+
+    try {
+      await locks.lock('test1');
+      await locks.unlock('test1');
+      await locks.lock('test2');
+      await locks.unlock('test2');
+    } catch (e) {
+      assert(false);
     }
-    function unlock(callback) {
-      return locks.unlock('test', callback);
+  });
+
+  it('should flunk a second lock by the same module', async function() {
+    const locks = apos.modules['apostrophe-locks'];
+
+    await locks.lock('test');
+
+    try {
+      await locks.lock('test');
+    } catch (e) {
+      assert(e);
```

Comment:
It's OK to do this assert but you haven't proven that an exception definitely *did* get thrown. The way to do that is to put assert(false) immediately after the `await locks.lock` call, i.e. where it should never be reached if the exception is thrown.

With that you actually shouldn't need the try/catch at all because what you really want to know is that you were stopped from continuing (by an exception).

Reply:
✔️ I'll add the `assert(false)`. If I don't catch anything it'll break subsequent tests, though.